aten/src/ATen/native/AveragePool2d.cpp

```diff
@@ -246,6 +264,11 @@ Tensor& avg_pool2d_backward_out_cpu_template(
   TORCH_CHECK((ndim == 3 || ndim == 4),
     "non-empty 3D or 4D (batch mode) tensor expected for input");
 
+  if (divisor_override.has_value()) {
+    TORCH_CHECK(divisor_override.value() != 0, "divisor must be not zero");
+    TORCH_CHECK(input.scalar_type() != ScalarType::Long, "avg_pool2d backward function with divisor is not supported for LongTensor");
```

Comment:
Backward requires division, so I do not support it for LongTensor and, as I understand, there is no way to get grad for LongTensors at all: 

>>> torch.tensor([1, 2, 3], dtype=torch.long, requires_grad=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: Only Tensors of floating point dtype can require gradients

Reply:
right, that's my point about checking that the type isn't long -- this function doesn't even work for long, right? (because it will fail in the dispatch macro), so why do additional checking specific to the divisor_override?

Reply:
Ok, agree.